packages/opentelemetry-context-async-hooks/src/AbstractAsyncHooksContextManager.ts

```diff
@@ -131,15 +138,22 @@ export abstract class AbstractAsyncHooksContextManager
    * @param ee EventEmitter instance
    * @param original reference to the patched method
    */
-  private _patchRemoveListener(ee: EventEmitter, original: Function) {
+  private _patchRemoveListener(
+    ee: EventEmitter,
+    original: RemoveListener
+  ): RemoveListener {
     const contextManager = this;
-    return function (this: never, event: string, listener: Func<void>) {
+    return function (
+      this: ThisParameterType<RemoveListener>,
+      ...args: Parameters<RemoveListener>
+    ): ReturnType<RemoveListener> {
+      const [event, listener] = args;
       const events = contextManager._getPatchMap(ee)?.[event];
-      if (events === undefined) {
-        return original.call(this, event, listener);
+      if (events) {
+        const patchedListener = events.get(listener);
+        args[1] = patchedListener || listener;
       }
-      const patchedListener = events.get(listener);
-      return original.call(this, event, patchedListener || listener);
+      return original.apply(this, args);
```

Comment:
I changed all the delegating calls in this file to take `...args` and pass them through to the original via `apply()`, mostly for consistency, but also arguably safer. If Node adds extra optional arguments in the future we will pass them through correctly.